### ISSUE TITLE
useNuxtApp().$vuetify is only available after the middleware

### ISSUE THREAD

**Issue by Dominic-Marcelino** [userquin/vuetify-nuxt-module#87]
Hi,
thanks for the great module! Currently evaluating to replace our custom integration with it. One problem I run into is that the plugin is only registered after the middleware.

https://github.com/userquin/vuetify-nuxt-module/blob/804abc7fe16ac69264eba0ad6163f9ba9089a77d/src/runtime/plugins/vuetify.ts#L5-L11

In our use-case we have some checkups in a global middleware that e.g checks for the user and hydrates the user-state which includes setting the theme:
```ts
const nuxtApp = useNuxtApp();
const vuetify = nuxtApp.$vuetify;
vuetify.global.name.value = theme;
```
This isn't possible with the module, as `nuxtApp.$vuetify` is still undefined at this point.

Is there any specific reason to only register it afterwards?
Looking forward to your feedback.

Thanks!

**Comment by userquin** [comment]
That plugin is only used when i18n module present. 

Maybe you can change the logic to restore the theme, check this entry in the docs using local storage https://vuetify-nuxt-module.netlify.app/guide/server-side-rendering.html#vuetify-themes

**Comment by userquin** [comment]
If you have any suggestion feel free to share the logic or send a PR.

**Comment by Dominic-Marcelino** [comment]
I played around a bit but unfortunately wasn't able to solve it without any other drawbacks. If I modified the module to register the plugin directly and synchronous it's been available from the middlewares, but lead to some hydration missmatches in the date modules.

- - -

> That plugin is only used when i18n module present.

The plugin is always available, it yust differentiates between a sync and async registration
https://github.com/userquin/vuetify-nuxt-module/blob/804abc7fe16ac69264eba0ad6163f9ba9089a77d/src/utils/configure-nuxt.ts#L79-L81

- - -
All in one It's been easier for me to refactor my app than to find a solution for this 😅
Feel free to close the issue if that's nothing you're planning to look into for now.

**Comment by userquin** [comment]
@Dominic-Marcelino can you provide a minimal repro? If you provide it I will try to find a solution in the afternoon

**Comment by Dominic-Marcelino** [comment]
Sure, when using the provided playground you can add the following middleware:

````ts 
// middlewares/0_test.global.ts

export default defineNuxtRouteMiddleware((to) => {
  if (process.client) {
    const nuxtApp = useNuxtApp()

    console.log('NUXT APP')
    console.log(nuxtApp)

    console.log('NuxtApp vuetify theme')
    console.log(nuxtApp.$vuetify.theme.current.value)

  }
})
```



In the `nuxtApp` log you can see that it'll be available afterwards, but won't be available yet on middleware execution:

![Bildschirmfoto 2023-08-28 um 11 48 15](https://github.com/userquin/vuetify-nuxt-module/assets/46492597/1b2bc684-79d2-432b-b00f-bfdf4aa00095)

**Comment by userquin** [comment]
@Dominic-Marcelino maybe you can register `app:created` hook and apply the logic inside the callback (check this middleware https://github.com/elk-zone/elk/blob/main/middleware/auth.ts using app:suspense hook in isHydrated prop)

**Comment by Dominic-Marcelino** [comment]
@userquin I already finished a refactoring in my app to refactor the theme from inside setup. This is just an example for the problem, in case you'd like to look into it on how to provide the plugin before. Otherwise the ticket can be closed. Sorry for being unclear.

**Comment by userquin** [comment]
@Dominic-Marcelino I'm preparing new PR to allow use vuetify in the middleware, requires Nuxt 3.9+ if using i18n module, I'm changing the logic of vuetify and vuetify.sync runtime plugins (main plugins):

![imagen](https://github.com/userquin/vuetify-nuxt-module/assets/6311119/7066864a-662d-4f6e-918e-ae5a7e095207)

**Comment by userquin** [comment]
@Dominic-Marcelino should be fixed in v0.9.0
